File: September-2025/06_09_2025_gfg.py
class Solution:
    def lengthOfLoop(self, head):
        """
        :type head: Node
        :rtype: int
        """
        slow = head
        fast = head

        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
            if slow == fast:
                break

        if not fast or not fast.next:
            return 0

        count = 1
        slow = slow.next
        while slow != fast:
            count += 1
            slow = slow.next

        return count

File: September-2025/06_09_2025_gfg.java
/*
class Node {
    int data;
    Node next;

    Node(int x) {
        data = x;
        next = null;
    }
}
*/

class Solution {
    public int lengthOfLoop(Node head) {
        // code here
        Node slow = head, fast = head;
        boolean loopExists = false;
        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
            if (slow == fast) {
                loopExists = true;
                break;
            }
        }

        if (!loopExists) {
            return 0;
        }

        int count = 1;
        Node current = slow;
        while (current.next != slow) {
            current = current.next;
            count++;
        }
        return count;
    }
}